spec/cat_spec.rb

```diff
@@ -0,0 +1,43 @@
+require 'cat'
+describe Cat do
+  subject(:cat) { described_class.new('tips/cat.txt') }
+  let!(:tips) { cat.tips }
+
+  describe 'initialisation' do
+    it 'throws an error if the tip file does not exist' do
+      expect { Cat.new('nonexistent_cd.txt') }.to raise_error(Errno::ENOENT)
+    end
+  end
+
+  describe '#tips' do
+    it 'loads the tips' do
+      expect(cat.tips.count).to eq(2)
+    end
+  end
+
+  describe '#suggest_tips' do
+    it 'passes all of the tips if none of the flags are used' do
+      expect(cat.suggest_tips('')).to eq([tips[:b], tips[:n]])
+    end
+
+    describe '-b flag sugestions' do
+      it 'suggests the -b tip when it is not used' do
+        expect(cat.suggest_tips('-n').include?(tips[:b])).to be true
```

Comment:
Expectations in arguments need to be how the car command would really be used i.e. with a filename. Looks like the way code is written a character of n in filename will break this test. Likewise lots of places.